core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -2350,6 +2350,16 @@ object SparkContext extends Logging {
     }
   }
 
+  private[spark] def getActiveContext(): Option[SparkContext] = {
+    SPARK_CONTEXT_CONSTRUCTOR_LOCK.synchronized {
+      Option(activeContext.get())
+    }
+  }
+
+  private[spark] def stopActiveContext(): Unit = {
```

Comment:
+1. I don't like stopping SparkContext before running tests, either. It will hide the mistakes in other tests.